assets/vue/components/signalement-form/SignalementFormAddress.vue

```diff
@@ -0,0 +1,176 @@
+      },
+      set (newValue: string) {
+        this.$emit('update:modelValue', newValue)
+      }
+    }
+  },
+  created () {
+    watch(
+      () => this.formStore.data[this.idAdress],
+      (newValue: any) => {
+        clearTimeout(this.idFetchTimeout)
+        this.idFetchTimeout = setTimeout(() => {
+          if (newValue.length > 10) {
+            requests.validateAddress(newValue, this.onAddressFound)
+          }
+        })
+      }
+    )
+  },
+  methods: {
+    updateValue (value: any) {
+      this.$emit('update:modelValue', value)
+    },
+    handleClickButton (type:string, param:string) {
+      if (this.clickEvent !== undefined) {
+        this.clickEvent(type, param)
+      }
+    },
+    handleSubscreenModelUpdate (newValue: string) {
+      // Mettre à jour la valeur dans formStore.data lorsque la valeur du sous-écran change
+      this.formStore.data[this.idSubscreen] = newValue
+    },
+    generateSubscreenData (id: string, data: any[]) {
+      return data.map((component) => {
+        return {
+          ...component,
+          slug: id + '_' + component.slug
+        }
+      })
+    },
+    onAddressFound (requestResponse: any) {
+      const container = document.querySelector('#sous_menu')
+      const subscreen = document.querySelector('#' + this.idSubscreen)
+      const buttonShow = document.querySelector('#' + this.idShow)
+
+      // TODO : que faire si code postal dans département non ouvert ?
+      // TODO : répertorier les exclusions de code postal du 69 ?
+      // TODO : vérifier si dans territoire expé NDE pour comportement différent ?
+      if (container) {
+        container.innerHTML = ''
+
+        for (const feature of requestResponse.features) {
+          const suggestion = document.createElement('div')
+          suggestion.classList.add(
+            'fr-col-12',
+            'fr-p-3v',
+            'fr-text-label--blue-france',
+            'fr-adresse-suggestion'
+          )
+          suggestion.innerHTML = feature.properties.label
+          suggestion.addEventListener('click', () => {
+            this.formStore.data[this.id + '_tous_les_champs_numero'] = feature.properties.name
+            this.formStore.data[this.id + '_tous_les_champs_code_postal'] = feature.properties.postcode
+            this.formStore.data[this.id + '_tous_les_champs_commune'] = feature.properties.city
+            this.formStore.data[this.id + '_tous_les_champs_insee'] = feature.properties.citycode
+            this.formStore.data[this.id + '_tous_les_champs_geoloc_lat'] = feature.geometry.coordinates[0]
+            this.formStore.data[this.id + '_tous_les_champs_geoloc_lng'] = feature.geometry.coordinates[1]
+
+            container.innerHTML = ''
+            if (subscreen) {
+              subscreen.classList.remove('fr-hidden')
+            }
+            if (buttonShow) {
+              buttonShow.classList.add('fr-hidden')
+            }
+          })
+          container.appendChild(suggestion)
```

Comment:
On peut pas rendre cela directement dans le template avec v-for sur les suggestions qui serait une donnée du composant et des booleans pour masquer subscreen et bouton ?

Reply:
je ne suis pas sûre de comprendre ce que tu veux faire. Seuls les 3 premières valeurs sont visibles dans le DOM, pas les 3 autres. Du coup je ne vois pas trop comment on gagnerait en simplicité

Reply:
done